src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComActivator.cs

```diff
@@ -177,6 +197,72 @@ private static Type FindClassType(Guid clsid, string assemblyName, string typeNa
             throw new COMException(string.Empty, CLASS_E_CLASSNOTAVAILABLE);
         }
 
+        private static AssemblyLoadContext GetALC(string assemblyPath)
+        {
+            AssemblyLoadContext alc;
+            return LazyInitializer.EnsureInitialized(ref s_AssemblyLoadContexts).TryGetValue(assemblyPath, out alc) ?
```

Comment:
Are there any interesting scenarios where ComActivator is used, but `s_AssemblyLoadContexts` will stay unitialized?

LazyInitializer here looks like overkill.

Reply:
I used this at the suggestion of @stephentoub. I originally was using the `Interlocked.Compare` method, but this seems less prone to mistakes. Is there a lot of overhead here?

Reply:
I think Jan's suggesting that lazy initialization in general here is overkill, that you could just initialize it as part of the declaration.

Reply:
As Stephen said.

Reply:
Oh I see - duh. That makes sense.